File: src/org/thoughtcrime/securesms/animation/AnimationCompleteListener.java
package org.thoughtcrime.securesms.animation;


import android.animation.Animator;

public abstract class AnimationCompleteListener implements Animator.AnimatorListener {
  @Override
  public final void onAnimationStart(Animator animation) {}

  @Override
  public abstract void onAnimationEnd(Animator animation);

  @Override
  public final void onAnimationCancel(Animator animation) {}
  @Override
  public final void onAnimationRepeat(Animator animation) {}
}

File: app/src/main/java/org/thoughtcrime/securesms/animation/AnimationCompleteListener.kt
/*
 * Copyright 2024 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.thoughtcrime.securesms.animation

import android.animation.Animator

abstract class AnimationCompleteListener : Animator.AnimatorListener {
  abstract override fun onAnimationEnd(animation: Animator)

  override fun onAnimationStart(animation: Animator) = Unit
  override fun onAnimationCancel(animation: Animator) = Unit
  override fun onAnimationRepeat(animation: Animator) = Unit
}
